perllib/FixMyStreet/Roles/Extra.pm

```diff
@@ -135,6 +135,49 @@ sub push_extra_fields {
     $self->extra({ %$extra, $META_FIELD => [ @$existing, @fields ] });
 }
 
+=head2 update_extra_field
+
+    $problem->update_extra_field( { ... } );
+
+Given an extra field, will replace one with the same code in the
+existing list of fields, or add to the end if not present.
+
+=cut
+
+sub update_extra_field {
```

Comment:
It'd be good to have some tests of these new functions. I think there's a bug where the extra fields are replaced entirely with the passed `$field` if it doesn't already exist - see other comment.